### PR TITLE
feat(storage): add integration test for universe domain

### PR DESCRIPTION
**Pull request by shubhangi-google**
Adding integration tests for Universe Domain feature
these tests will run with test project credentials

### CODE REVIEW

File: google-cloud-storage/acceptance/storage_helper.rb [hunk 1]
```diff
@@ -28,6 +28,12 @@
   Minitest::Reporters.use! [Minitest::Reporters::SpecReporter.new, Minitest::Reporters::JUnitReporter.new]
 end
 
+# Universe Domain Test project Credentials
+$test_universe_domain = ENV["TEST_UNIVERSE_DOMAIN"]
```
Comment (bajajneha27) [inline google-cloud-storage/acceptance/storage_helper.rb:32]: Let's not keep these in the common file and move it to the new file created.

**Comment by bajajneha27** [comment]
@shubhangi-google the acceptance test is failing, can you please check?

**Comment by shubhangi-google** [comment]
> @shubhangi-google the acceptance test is failing, can you please check?

hi @bajajneha27 
earlier kokoro was failing with a test failure now after resubmitting the PR
it is failing with- **There was insufficient space**
https://btx.cloud.google.com/invocations/10d0b291-1669-48fb-9bc6-fa5037342fc2/targets/cloud-devrel%2Fclient-libraries%2Fgoogle-cloud-ruby%2Fpresubmit%2Facceptance/log

please suggest how to proceed

File: google-cloud-storage/acceptance/storage/universe_domain_test.rb [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require "storage_helper"
+
+describe Google::Cloud::Storage do
+
+  # Universe Domain Test project Credentials
+  $test_universe_domain = ENV["TEST_UNIVERSE_DOMAIN"]
```
Comment (bajajneha27) [inline google-cloud-storage/acceptance/storage/universe_domain_test.rb:20]: Where are we setting these environment variables?

Comment (shubhangi-google) [reply google-cloud-storage/acceptance/storage/universe_domain_test.rb:20]: as per our discussion on call we are not setting these env vars explicitly
the tests were passing without explicit declaration only one test was failing related to universe domain location

Comment (shubhangi-google) [reply google-cloud-storage/acceptance/storage/universe_domain_test.rb:20]: shall I add the explicit declaration in integration.sh file?

Comment (bajajneha27) [reply google-cloud-storage/acceptance/storage/universe_domain_test.rb:20]: Yes. We won't be able to get their values otherwise.

Comment (shubhangi-google) [reply google-cloud-storage/acceptance/storage/universe_domain_test.rb:27]: hi @bajajneha27 
still  have the following failure in one of my test case
I have set all the system vars in test.rb file only then also I guess test_location var is missing in env vars
please suggest how to proceed with this 

  1) Failure:
Google::Cloud::Storage#test_0001_creates a new bucket with universe_domain [acceptance/storage/universe_domain_test.rb:61]:
Expected: nil
  Actual: "US"

163 runs, 1302 assertions, 1 failures, 0 errors, 2 skips

**Comment by bajajneha27** [comment]
The integration test is still failing. Can you please check?

File: google-cloud-storage/acceptance/storage/universe_domain_test.rb (hunk 2 above)
Comment (bajajneha27) [reply google-cloud-storage/acceptance/storage/universe_domain_test.rb:27]: I'd check if I'm getting data from the secret manager. Try moving these variables to integration.sh and double all the key names / paths are same.

Comment (shubhangi-google) [reply google-cloud-storage/acceptance/storage/universe_domain_test.rb:27]: I had tried putting theses in integration.sh
but after that none of the values were being fetched from env vars

File: google-cloud-storage/acceptance/storage/universe_domain_test.rb [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require "storage_helper"
+
+describe Google::Cloud::Storage do
+
+  system(
+    "export TEST_UNIVERSE_DOMAIN_CREDENTIAL=$(realpath ${KOKORO_GFILE_DIR}/secret_manager/client-library-test-universe-domain-credential)
```
[the remark this answers is not in the document]
Comment (bajajneha27) [reply google-cloud-storage/acceptance/storage/universe_domain_test.rb:20]: Right now, we're using bash code inside Ruby script. Can we instead use Ruby commands to set the variables?

**Comment by danielduhh** [comment]
@dazuma do you have a preference in how the secrets are handled for Ruby?

**Comment by bajajneha27** [comment]
@shubhangi-google can you please resolve the conflict here so that I can merge the PR?

**Pull request merged**